Update COOP/COEP article (#4042)
GoogleChrome/web.dev: src/site/content/en/blog/coop-coep/index.md

feedback:
  - api
---
{% Aside 'gotchas' %}
**[15th Oct. 2020]**: `self.crossOriginIsolated` and Shared Array Buffer on Android

> Chrome are now available in Chrome 87. `document.domain` is now immutable when

Reviewer: SABs will be launched in M88 not M87 on Android, no changes for Desktop yet.

Author: This should have been sorted. Please check.

> `self.crossOriginIsolated` returns `true`. `performance.measureMemory` in on
> origin trial from Chrome 87.

Reviewer 2: ```suggestion
`self.crossOriginIsolated` returns `true`. `performance.measureMemory` in on
origin trial from Chrome 87.
```
```suggestion
`self.crossOriginIsolated` returns `true`. `performance.measureMemory()` is
ending its origin trial and will be enabled by default in Chrome 88.
```

Author: Are we sure `performance.measureMemory` will be also shipped in Chrome 88?

Reviewer: @ulan AFAIK the OT for performance.measureMemory() is ending and the API will be shipped gated behind COOP/COEP in M88, correct?
@agektmr enabled by default gated behind COOP/COEPm which is a change to the OT as the OT was not gated yet

Author: I can't find Intent to Ship for performance.measureMemory(). Without that I'm not sure if that's something we should say on a blog post.

Reviewer: No I2S yet send. I've kicked off a thread with ulan to discuss earlier to day, will add you to that one.

When are you planning to release the blog post?

Author: Ideally before 15th (Thursday) given the beta announcement article goes out on that day.

Reviewer: As discussed offline: From my point of view we can mention the API already in the blog post (OT ending in M87 and planned to ship in M88)

Reviewer 2: > Ideally before 15th (Thursday) given the beta announcement article goes out on that day.

The beta announcement goes up as soon as I see the beta on the dashboards. That usually happens around 11:00 am PST.

Author: Please check once again: https://deploy-preview-4042--web-dev-staging.netlify.app/coop-coep/
Sqoosh's adoption is exciting and would love to add somewhere in the article but I can't find a good place. Any suggestions?

Reviewer: Right, that's not easy.

What about at the end of the section 'Deploy COOP and COEP to make your website cross-origin isolated' pointing to the example?

Author: @lutzvahl Please check how it looks: https://github.com/GoogleChrome/web.dev/pull/4042/files#diff-5afd2bd259d9f1c14af088410be752de83c1eb1a05e55e8fae8cd0bb0d1df7e3R181

https://deploy-preview-4042--web-dev-staging.netlify.app/coop-coep/#determine-whether-isolation-succeeded-with-self.crossoriginisolated

Reviewer: I would adjust the text to:
squoosh.app now uses COOP / COEP to gain access to Wasm Threads (and Shared Array Buffer) as well on Android Chrome.

Position and styling looks good!

Author: https://github.com/GoogleChrome/web.dev/pull/4042/commits/ad6f9fcfe95f39da5b35c2227e251e653a324459
Thank you for your review!

> **[1st Sep. 2020]**: COOP Reporting is now available behind a flag in Chrome 86.

Reviewer 2: ```suggestion
**[1st Sep. 2020]**: COOP Reporting is behind the `#cross-origin-isolated`
flag as of Chrome 86.
```

Added instructions.

{% endAside %}

Some web APIs increase the risk of side-channel attacks like Spectre. To
mitigate that risk, browsers offer an opt-in-based isolated environment called
cross-origin isolated. With a cross-origin isolated state, the webpage will be
able to use privileged features including:

* [`SharedArrayBuffer`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer)

>   (required for WebAssembly Threads) (Available from Chrome 87)

Reviewer: See above, SABs will be shipped in M88 on Android

feedback:
  - api
---

> {% Aside 'gotchas' %}
> 
> **[15th Oct. 2020]**: `self.crossOriginIsolated` is available from Chrome 87.
> Reflecting that, `document.domain` is immutable when `self.crossOriginIsolated`
> returns `true`. `performance.measureMemory` ending its origin trial and is
> planned to be enabled by default in Chrome 88. Shared Array Buffer on Android
> Chrome will be available from Chrome 88.
> 
> **[1st Sep. 2020]**: COOP Reporting is behind the `#cross-origin-isolated` flag
> in Chrome 86. Added instructions.
> 
> {% endAside %}

Reviewer 3: ```suggestion
{% Banner 'caution', 'body' %}

## Updates

### October 15th, 2020

`self.crossOriginIsolated` is available from Chrome 87.
Reflecting that, `document.domain` is immutable when `self.crossOriginIsolated`
returns `true`. `performance.measureMemory` ending its origin trial and is
planned to be enabled by default in Chrome 88. Shared Array Buffer on Android
Chrome will be available from Chrome 88.

### September 1st, 2020

COOP Reporting is behind the `#cross-origin-isolated` flag
in Chrome 86.

{% endBanner %}
```

I'm not seeing any instructions for the `#cross-origin-isolated` on this page. Also, please avoid `Added instructions` and use our common phrasing: `See [Section name](#link)`. 

Let's try using a body banner here rather than a gotcha: https://web.dev/handbook/web-dev-components/#body-banners

LMK if you want me to explain why a gotcha doesn't seem appropriate here

feedback:
  - api
---
{% Banner 'caution', 'body' %}

**Updates**

**October 15th, 2020**: `self.crossOriginIsolated` is available from Chrome 87.
Reflecting that, `document.domain` is immutable when `self.crossOriginIsolated`
returns `true`. `performance.measureMemory` ending its origin trial and is
planned to be enabled by default in Chrome 88. Shared Array Buffer on Android
Chrome will be available from Chrome 88.

**September 1st, 2020**: COOP Reporting is behind the `#cross-origin-isolated` flag

> in Chrome 86.

Reviewer 3: @agektmr you previously said "Added instructions" here but I'm not seeing any instructions in this post. Also, please use our standard phrasing `See [X](link)` rather than `Added instructions`.

Author: This is the change I made for this: https://github.com/GoogleChrome/web.dev/commit/de54c4300e3d5fd76f424eb83e339a2a68d2b8d3#diff-5afd2bd259d9f1c14af088410be752de83c1eb1a05e55e8fae8cd0bb0d1df7e3
I wanted to link to the line starting with "The COOP Reporting API in Chrome is available after version 86 with one of two conditions:" but unable to do it because it's part of summary.

Author: I tried to add `{: #coop-reporting}` at inside `{% DetailsSummary %}` but proved it doesn't work. Please don't worry. This isn't a big deal.

Reviewer 3: Ah, OK. In that case I think we can just use a [named anchor](https://stackoverflow.com/a/10510353/1669860). I'll add it now and see if it works.

> {% endBanner %}

Reviewer 3: The banner has more top and bottom padding than I'd like but I still think it's more appropriate than the Gotchas aside:

![image](https://user-images.githubusercontent.com/4713486/96033748-5e0dd900-0e15-11eb-8dd7-b17f038d1471.png)

See previous comment: https://github.com/GoogleChrome/web.dev/pull/4042#pullrequestreview-508634911

feedback:
  - api
---
{% Banner 'caution', 'body' %}

> ## Updates
> 
> ### October 15th, 2020
> 
> `self.crossOriginIsolated` is available from Chrome 87.
> Reflecting that, `document.domain` is immutable when `self.crossOriginIsolated`
> returns `true`. `performance.measureMemory` ending its origin trial and is
> planned to be enabled by default in Chrome 88. Shared Array Buffer on Android
> Chrome will be available from Chrome 88.
> 
> ### September 1st, 2020
> 
> COOP Reporting is behind the `#cross-origin-isolated` flag
> in Chrome 86.

Reviewer 3: ```suggestion
**Updates**

**October 15th, 2020**: `self.crossOriginIsolated` is available from Chrome 87.
Reflecting that, `document.domain` is immutable when `self.crossOriginIsolated`
returns `true`. `performance.measureMemory` ending its origin trial and is
planned to be enabled by default in Chrome 88. Shared Array Buffer on Android
Chrome will be available from Chrome 88.

**September 1st, 2020**: COOP Reporting is behind the `#cross-origin-isolated` flag
in Chrome 86.
```

parameters](https://first-party-test.glitch.me/coep).
{% endAside %}

{% Aside %}

> [squoosh.app](https://squoosh.app) [now uses COOP /

Reviewer 3: ```suggestion
[Squoosh](https://squoosh.app) (an image optimization PWA) [now uses COOP /
```
It's a bit awkward to have two links right next to each other so I'm just adding some text to add space between the links
